api/handlers.go

```diff
@@ -384,17 +378,21 @@ func (api *API) userGET(w http.ResponseWriter, req *http.Request, _ httprouter.P
 
 // userLimitsGET returns the speed limits which apply to this user.
 func (api *API) userLimitsGET(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
-	t, err := tokenFromRequest(req)
-	if err != nil {
-		api.WriteJSON(w, database.UserLimits[database.TierAnonymous])
-		return
+	// First check for an API key.
+	ak, err := apiKeyFromRequest(req)
```

Comment:
Once we have middleware this can probably also be moved to that.

e.g. we could have a `mustAuthenticate` and a `canAuthenticate` middleware. This would use `canAuthenticate` and would potentially pass a nil user that you can check for to see if the user was either loged in via api key or token.

Reply:
`withAuth` is already a middleware, so we have that. We additionally call `apiKeyFromRequest` from `userLimitsGET` because we want to specifically not use the middleware for that endpoint because we want it to handle both authenticated and unauthenticated calls.

Reply:
F/U: https://github.com/SkynetLabs/skynet-accounts/issues/127